addons/website_sale/controllers/main.py

```diff
@@ -402,39 +362,31 @@ def checkout_redirection(self, order):
             return request.redirect('/shop')
 
         # if transaction pending / done: redirect to confirmation
-        tx = context.get('website_sale_transaction')
+        tx = request.env.context.get('website_sale_transaction')
         if tx and tx.state != 'draft':
             return request.redirect('/shop/payment/confirmation/%s' % order.id)
 
     def checkout_values(self, data=None):
-        cr, uid, context, registry = request.cr, request.uid, request.context, request.registry
-        orm_partner = registry.get('res.partner')
-        orm_user = registry.get('res.users')
-        orm_country = registry.get('res.country')
-        state_orm = registry.get('res.country.state')
-
-        country_ids = orm_country.search(cr, SUPERUSER_ID, [], context=context)
-        countries = orm_country.browse(cr, SUPERUSER_ID, country_ids, context)
-        states_ids = state_orm.search(cr, SUPERUSER_ID, [], context=context)
-        states = state_orm.browse(cr, SUPERUSER_ID, states_ids, context)
-        partner = orm_user.browse(cr, SUPERUSER_ID, request.uid, context).partner_id
+        countries = request.env['res.country'].sudo().search([])
+        states = request.env['res.country.state'].sudo().search([])
+        partner = request.env.user.partner_id.sudo()
```

Comment:
actually no need for sudo here because `self.env.user` is already [browsed as superuser](https://github.com/odoo/odoo/blob/f16c5aacf1651b29cc003d354e913c3244d15458/openerp/api.py#L795)